flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskExecutor.java

```diff
@@ -1368,7 +1383,16 @@ private void disconnectAndTryReconnectToJobManager(
             JobTable.Connection jobManagerConnection, Exception cause) {
         try (MdcCloseable ignored =
                 MdcUtils.withContext(MdcUtils.asContextData(jobManagerConnection.getJobId()))) {
-            disconnectJobManagerConnection(jobManagerConnection, cause);
+            disconnectJobManagerConnection(jobManagerConnection, cause, false);
+            jobLeaderService.reconnect(jobManagerConnection.getJobId());
+        }
+    }
+
+    private void disconnectAndTryReconnectToJobManagerPossibleRetainPartitions(
```

Comment:
I prefer to name it `disconnectAndTryReconnectToJobManager` and accept a param `releasePartitions`.
It can be reused by the existing `disconnectAndTryReconnectToJobManager()` method.